### DISCUSSION TITLE
Luks encryption, Display issues

### DISCUSSION THREAD

**Discussion by BrainWart** [BrainWart/x13s-nixos#4] [General]
@foorschtbar 
> It feels wrong to open up issues in order to talk about something.
> 
> After got a working luks setup my screen now goes of after some time. even in luks passpharse screen or after stage 2, when login screen appears - depends how fast i am.

I am not using LUKS. I am working on a build which will be based on the latest linux kernel and am interested in seeing if there are any updates in the display area. I am currently running windows. Linux was unable to drive multiple external displays on this laptop. If it boots, I will create a branch based on the changes I have made. As I write this, the kernel is building and I will write back when I get something working. It's possible, your issues may be resolved as well. Are you able to share your configuration?

**Comment by foorschtbar** [comment]
Sounds promising! 

I don't think, the LUKS is the problem. The screen still turn off after a few seconds - already in stage 1. 

## bootloader.nix
- is 1:1 yours

## configuration.nix
- changed `hostname`, `timezone`
- removed `services.printing`
- changed username, added `users.mutableUsers = false;`

## hardware-configuration.nix:
```nix
# Do not modify this file!  It was generated by ‘nixos-generate-config’
# and may be overwritten by future invocations.  Please make changes
# to /etc/nixos/configuration.nix instead.
{ config, lib, pkgs, modulesPath, ... }:

{
  imports =
    [ (modulesPath + "/installer/scan/not-detected.nix")
    ];

  boot.initrd.availableKernelModules = [ "nvme" "usb_storage" ];
  boot.initrd.kernelModules = [ "dm-snapshot" ];
  boot.kernelModules = [ ];
  boot.extraModulePackages = [ ];
  boot.initrd.luks.devices."root".device = "/dev/disk/by-uuid/8a9dbd7b-e0ff-45ea-9566-1b4b059f2fb4";

  fileSystems."/" =
    { device = "/dev/disk/by-uuid/d458d32e-ee0b-4fa0-b787-5f191bb27d2a";
      fsType = "ext4";
    };

  fileSystems."/boot" =
    { device = "/dev/disk/by-uuid/A9AA-B34E";
      fsType = "vfat";
    };

  swapDevices =
    [ { device = "/dev/disk/by-uuid/91c1c386-2751-4a5c-9467-7881013ff3d8"; }
    ];

  # Enables DHCP on each ethernet and wireless interface. In case of scripted networking
  # (the default) this is the recommended approach. When using systemd-networkd it's
  # still possible to use this option, but it's recommended to use it in conjunction
  # with explicit per-interface declarations with `networking.interfaces.<interface>.useDHCP`.
  networking.useDHCP = lib.mkDefault true;
  # networking.interfaces.enx0826ae3df5a1.useDHCP = lib.mkDefault true;
  # networking.interfaces.wlP6p1s0.useDHCP = lib.mkDefault true;

  nixpkgs.hostPlatform = lib.mkDefault "aarch64-linux";
}

```

**Comment by BrainWart** [comment]
No luck yet. I was trying to build an iso using https://github.com/nix-community/nixos-generators and it doesn't seem to want to boot it when I used Rufus to create a live USB. I'll look more as I get time

**Comment by BrainWart** [comment]
@foorschtbar

I finally made some time to mess with this these past few days. I have been able to successfully build the image and have it boot from USB. The latest commit on that branch uses `linux_latest` which may not be suitable for this machine. It boots for me but there are some issues. No battery meter, and USB seems broken. USB may be broken because of the module which was disabled to ensure USB boot would be successful. I do not have the time to image my machine but, I will post back is removing that like seems to fix things when I get around to it. There is a commit which uses jhovold's [linux 6.7 branch](https://github.com/jhovold/linux/wiki/X13s) as well. It may work better. I got tired of the long builds. If you get the time, lmk how it's working for you.

https://github.com/BrainWart/x13s-nixos/tree/flake
